src/engine_openssl.c

```diff
@@ -331,6 +349,30 @@ static void tls_free_cert(tls_cert *cert) {
     *cert = NULL;
 }
 
+static void tls_set_alpn_protocols(void *ctx, const char **protos, int len) {
+    struct openssl_ctx *c = ctx;
+
+    if (c->alpn_protocols) {
+        free(c->alpn_protocols);
+    }
+
+    size_t protolen = 0;
+    for (int i=0; i < len; i++) {
+        protolen = strlen(protos[i]) + 1;
```

Comment:
looking at this again - why do this in a loop (multip assignments of protolen)? is this meant to be += here?

Reply:
correct